### PR TITLE
feat: improve shell cli for optional args [DET-3883] [DET-3886]

### PR DESCRIPTION
**Pull request by rb-determined-ai**
BREKING CHANGE: Change the format of the ssh options arguments to
`det shell start` and `det shell open`.  Instead of calling:

    det shell start --ssh-opt "-XY -o SomeOption='some string'"

which has multiple layers of quotes due to a two-step shell tokenizing,
now users should call:

    det shell start -- -XY -o SomeOption="some string"

This change also allows us to clean up our internal implementation to
avoid relying on shell tokenization internally.

### CODE REVIEW

File: cli/determined_cli/shell.py [hunk 1]
```diff
@@ -71,17 +71,32 @@ def _open_shell(master: str, shell: Command, additional_opts: str) -> None:
         if host == LOOPBACK_ADDRESS:
             host = "localhost"
 
+        # Use determined_cli.tunnel as a portable script for using the HTTP CONNECT mechanism,
+        # similar to `nc -X CONNECT -x ...` on unix systems.
```
Comment (dzhu) [inline cli/determined_cli/shell.py:75]: highly non-blocking: netcat has been ported to Windows, so I'm guessing this statement could technically be a bit broader. Also "unix" → "Unix".

**Comment by dzhu** [comment]
Also, commit/PR message: "BREKING" → "BREAKING".

**Comment by rb-determined-ai** [comment]
Also changing this PR to inject additional arguments after the ssh host when invoking ssh to solve DET-3886.

File: cli/determined_cli/shell.py (hunk 1 above)
Comment (rb-determined-ai) [reply cli/determined_cli/shell.py]: fixed.

**Comment by dzhu** [comment]
Looks good!

**Pull request merged**